account_payment_term_partner_holiday/models/account_invoice.py

```diff
@@ -17,12 +17,12 @@ def _onchange_payment_term_date_invoice(self):
 
     @api.onchange("date_due")
     def _onchange_date_due_account_payment_term_partner_holiday(self):
-        if self.date_due and self.payment_term_id and self.partner_id:
+        if self.date_due and self.partner_id:
             new_date_due = self.date_due
             is_date_in_holiday = self.partner_id.is_date_in_holiday(new_date_due)
-            if is_date_in_holiday:
-                res_date_end = is_date_in_holiday[1]
-                new_date_due = res_date_end + relativedelta(days=1)
+            while is_date_in_holiday:
```

Comment:
Why coming back again to a loop? This kind of loops are always dangerous.

Reply:
And why not having this stuff in a method for being shared in all code? Instead of `is_date_in_holiday`, put the method `_get_valid_due_date` or similar in the partner and given a date, return the next valid date, and call it from both places.

Reply:
Loop is necessary because:
- If actually `is_date_in_holiday()` function get first holiday record (with date_from and date_to), we will use date_to + 1 day to new date_due but, what happen if new date is too unavailable according to holidays?
- If `is_date_in_holiday()` function would return last date_to from holiday is absolutely incorrect.

IMO loop is the only way to get first new date due available according holidays.
What do you think about it?
It's correct "replace"  `is_date_in_holiday` to `_get_valid_due_date` to make loop inside and return only first valid date?

Reply:
OK about a loop, but not going on steps of 1 day. Just set next day as holidays end + 1, and repeat the check.

Reply:
Ok, but we keep loop or is better create `_get_valid_due_date()` function to do it?

Reply:
Create the function for not doing 2 times the same thing according my comment in https://github.com/OCA/account-payment/pull/429#discussion_r669337386